src/goto-programs/goto_convert.cpp

```diff
@@ -1830,8 +1830,13 @@ bool goto_convertt::get_string_constant(
       forall_operands(it, index_op)
         if(it->is_constant())
         {
-          unsigned long i = integer2ulong(
-            bv2integer(id2string(to_constant_expr(*it).get_value()), true));
+          mp_integer int_value;
+          if(to_integer(to_constant_expr(*it), int_value))
+            return true;
```

Comment:
The limitation of GitHub's suggestions feature (I have already provided feedback to them): my suggestion was meant to replace the entire bit of code, the lines above need to be removed now.